src/main/java/de/tum/in/www1/artemis/service/exam/ExamService.java

```diff
@@ -341,16 +367,237 @@ public ExamScoresDTO calculateExamScores(Long examId) {
+                    .orElseThrow(() -> new EntityNotFoundException("No student exam found for examId " + examId + " and userId " + studentId));
+
+            StudentExamWithGradeDTO studentExamWithGradeDTO = getStudentExamGradesForSummaryAsStudent(targetUser, studentExam);
+            var studentResult = studentExamWithGradeDTO.studentResult();
+            return Map.of(studentId, new BonusSourceResultDTO(studentResult.overallPointsAchieved(), studentResult.mostSeverePlagiarismVerdict()));
+        }
+        var scores = calculateExamScores(examId);
+        var studentIdSet = new HashSet<>(studentIds);
+        return scores.studentResults().stream().filter(studentResult -> studentIdSet.contains(studentResult.userId())).collect(Collectors.toMap(ExamScoresDTO.StudentResult::userId,
+                studentResult -> new BonusSourceResultDTO(studentResult.overallPointsAchieved(), studentResult.mostSeverePlagiarismVerdict())));
+
+    }
+
+    /**
+     * Return student exam result, aggregate points, assessment result for a student exam and grade calculations
+     * if the exam is assessed.
+     *
+     * See {@link StudentExamWithGradeDTO} for more explanation.
+     *
+     * @param targetUser the user who submitted the studentExam
+     * @param studentExam the student exam to be evaluated
+     * @return the student exam result with points and grade
+     */
+    public StudentExamWithGradeDTO getStudentExamGradesForSummaryAsStudent(User targetUser, StudentExam studentExam) {
+
+        loadQuizExercisesForStudentExam(studentExam);
+
+        // check that the studentExam has been submitted, otherwise /student-exams/conduction should be used
+        if (!studentExam.isSubmitted() || !studentExam.areResultsPublishedYet()) {
+            throw new AccessForbiddenException("You are not allowed to access the grade summary of a student exam which was NOT submitted!");
+        }
+
+        // fetch participations, submissions and results and connect them to the studentExam
+        fetchParticipationsSubmissionsAndResultsForRealExam(studentExam, targetUser);
+
+        List<StudentParticipation> participations = studentExam.getExercises().stream().flatMap(exercise -> exercise.getStudentParticipations().stream()).toList();
+        // fetch all submitted answers for quizzes
+        submittedAnswerRepository.loadQuizSubmissionsSubmittedAnswers(participations);
+
+        return calculateStudentResultWithGradeAndPoints(studentExam, participations);
+    }
+
+    /**
+     * Loads the quiz questions as is not possible to load them in a generic way with the entity graph used.
+     * See {@link StudentParticipationRepository#findByStudentExamWithEagerSubmissionsResult}
+     *
+     * @param studentExam the studentExam for which to load exercises
+     */
+    public void loadQuizExercisesForStudentExam(StudentExam studentExam) {
+        for (int i = 0; i < studentExam.getExercises().size(); i++) {
```

Comment:
This could benefit from a enhanced for loop
`for (Exercise exercise: studentExam.getExercises())`

Reply:
I only moved the method from [StudentExamResource](https://github.com/ls1intum/Artemis/pull/5668/files#diff-8098ab9f22aba5ddb5a01571b4841f329ac30ab0ff3bce8d32223f8b1c991b6fL843-L856) without modifying it. Besides, the value of `i` is used inside the method as well so in my opinion keeping it would be more idiomatic that using an enhanced loop + incrementing an integer.

Reply:
You are just adding it to a list aren't you? If you are, then setting it to an incrementing index is the same as using the add function

Reply:
It's not adding but replacing the elements at the index `i` as far as I understood.